FLEDGE_Key_Value_Server_trust_model.md

```diff
@@ -0,0 +1,171 @@
+5. The server looks up the matching data for the keys and returns it also in encrypted form.  [See the API explainer for details.](https://www.google.com/url?q=https://github.com/WICG/turtledove/blob/main/FLEDGE_Key_Value_Server_API.md&sa=D&source=docs&ust=1645211574244509&usg=AOvVaw3b_-LVRlUFTygnkUfCapYB)
+
+
+# Privacy and security considerations
+
+
+## Overall flow
+
+The FLEDGE explainer outlines the following security/privacy goals for the key-value servers:
+
+_The browser needs to trust that the server's return value for each key will be based only on that key and the hostname, and that the server does no event-level logging and has no other side effects based on these requests._
+
+To meet these requirements, the proposed design for the key-value server relies on a combination of:
+
+
+
+*   Attestation ensures that an adtech-operated key/value service runs an approved codebase.
+*   Data in use:
+    *   Lookup of key/value data happens within a secure and isolated Trusted Execution Environment. This prevents any party from learning the keys being requested.
+*   Data in transit:
+    *   Client devices will pre-load and periodically refresh the public encryption key signed by a KMS that the client devices recognize.
+    *   The KMS only grants the private key to successfully attested servers. 
+*   Data at rest:
+    *   The server will never do per-request lookups to persistent storage for the key value data. Techniques such as pre-loading the dataset should be used.
+*   Auditability:
+    *   Open source implementations of the key/value service ensure that these systems are publicly accessible and can be inspected by a broad set of stakeholders.
+
+
+
+![alt_text](assets/fledge_key_value_server.png "FLEDGE Key Value Server system diagram")
+
+
+
+## Threats
+
+Ideally, the key-value server would never get access to any information that could be used to identify a specific client or user. However, in practice, request timestamps and other necessary metadata included in lookup requests make this idealized system impractical. 
+
+The main threats that we’re concerned with are:
+
+
+
+1. That the timestamp of the request could be used to correlate the lookup keys with the request of the top-level page and thus identify a user. This is mitigated by limiting the logs that are written.
+2. During the FLEDGE auction, a client asks a key/value server about an assortment of keys that were previously stored on the client — perhaps including keys that were stored in multiple different contexts (different sites or different apps, via [Shared Storage](https://github.com/pythagoraskitty/shared-storage)).  That set of keys, therefore, risks leaking some information about the client's activity.  The privacy goal of key/value server design is to give the client confidence that the set of keys cannot be used for tracking or profiling purposes.  This threat is mitigated by the TEE protections to allow only pre-approved code.
```

Comment:
I don't think Shared Storage is relevant here.  The keys might come from different contexts because the same Interest Group `owner` might have called `joinAdInterestGroup()` (and set lookup keys) on multiple different domains.

Reply:
Thanks, I've removed the reference to Shared Storage there.